src/prpy/util.py

```diff
@@ -959,6 +959,27 @@ def CheckJointLimits(robot, q):
             description='position')
 
 
+def GetForwardKinematics(robot, q):
+    """
+    Get the forward kinematics for a specific joint configuration.
+
+    @param openravepy.robot robot: The robot object.
+    @param list             q:     List or array of joint positions.
+
+    @returns T_ee: The pose of the end effector (or last link in the
+                   serial chain) as a 4x4 matrix.
+    """
+    T_ee = None
+
+    # Save the robot state
+    sp = openravepy.Robot.SaveParameters
+    with robot.CreateRobotStateSaver():
```

Comment:
It would be better to only pass the `LinkTransformations` option here. Otherwise we'll save a bunch of extra state that we don't modify (e.g. joint limits, active DOFs, etc).